### PR TITLE
Add more code samples and updated information for NextJS and NextAuth

### PR DESCRIPTION
**Pull request by turnipdabeets**
### Your checklist for this pull request

Thank you for your contribution to the `next-dev-docs` repo. 

🚨Please review the [Pull request guidelines](../contributor-guide/contributor-guide.md#pull-request-guidelines) to this repository before submitting this PR.

- [ x] Your code builds clean without any errors or warnings
  
  ```bash
  yarn build
  ```

- [ x] You are using approved terminology

  Refer to the [style guide](../contributor-guide/style-guide.md) for help.

### Description 

This PR fills out the `Integrate with NextAuth` and `Getting Started with NextJS` guides with more code samples, updated information, and filling out highlighted questions. Also in this PR is deleting an unused nextauth file.

### CODE REVIEW

File: docs/get-started/getting-started-nextjs.mdx [hunk 1]
```diff
@@ -170,66 +165,70 @@ Don't forget, if you get stuck, let us know in our **[Slack community](https://j
...
+| ----------------------------- | ------------------------------------------------------------------------------------------------------------------------ |
+| NEXTAUTH_URL                  | Use the default value of http://localhost:3000                                                                           |
+| NEXTAUTH_SECRET               | generate one here: https://generate-secret.vercel.app/32                                                                 |
+| BEYOND_IDENTITY_CLIENT_ID     | Copy your app’s **Client ID** from the **External Protocol** tab on the app’s configuration page in your tenant          |
+| BEYOND_IDENTITY_CLIENT_SECRET | Copy your app’s **Client Secret** from the **External Protocol** tab on the app’s configuration page in your tenant      |
+| BEYOND_IDENTITY_DISCOVERY     | Copy your app’s **Discovery Endpoint** from the **External Protocol** tab on the app’s configuration page in your tenant |
+
+copy and paste the value from **Applications > {New Application} > External Protocol > Issuer**.
```
Comment (Will5) [inline docs/get-started/getting-started-nextjs.mdx:185]: nit: copy should be capitalized

File: docs/get-started/getting-started-nextjs.mdx [hunk 2]
```diff
@@ -170,66 +165,70 @@ Don't forget, if you get stuck, let us know in our **[Slack community](https://j
...
+
+Inside the `providers` array, add the following entry:
 
 ```javascript
 {
     id: "beyondidentity",
     name: "Beyond Identity",
     type: "oauth",
-    wellKnown: `https://auth-${process.env.REGION}.beyondidentity.com/v1/tenants/${process.env.TENANT_ID}/realms/${process.env.REALM_ID}/applications/${process.env.APPLICATION_ID}/.well-known/openid-configuration`,
+    wellKnown: process.env.BEYOND_IDENTITY_DISCOVERY,
```
Comment (Will5) [inline docs/get-started/getting-started-nextjs.mdx:228]: I love the simplification here 🙂

File: docs/get-started/getting-started-nextjs.mdx (hunk 1 above)
Comment (turnipdabeets) [reply docs/get-started/getting-started-nextjs.mdx]: updated!

File: docs/integration-guides/next-auth.md [hunk 3]
```diff
@@ -281,13 +391,15 @@ const BIBindPasskey = () => {
...
+**What's happening here?**
 
-1. The `useEffect` is only called once on page load. In this function, we initialize the Beyond Identity SDK and use `embedded.isBindPasskeyUrl` to check if the current page that was redirected to is in fact a valid `bind` URL.
+1. After tapping the email sent to the user's device, the user is directed to your application's `/bind` route appending to the `invoke_url` configured in the application config.
 
-2. If the URL is valid, we pull the URL using `window.location.href` and pass that directly into `embedded.bindPasskey` to complete the binding process.
+2. The `useEffect` is only called once on page load. In this function, we initialize the Beyond Identity SDK and use `embedded.isBindPasskeyUrl` to check if the current page that was redirected to is in fact a valid `bind` URL.
 
-3. Finally, the response of `embedded.bindPasskey` contains a `passkey` object, which represents the passkey bound to the device.
+3. If the URL is valid, we pull the URL using `window.location.href` and pass that directly into `embedded.bindPasskey` to complete the binding process.
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:400]: Is there anything actionable in this sentence? If not, I would indent it under the first step.

Also, be careful with the use 'we'. If the user is doing something in this step, say 'you' and put the verb (action) at the beginning, so when they scan the content they'll know what this step does quickly.

File: docs/integration-guides/next-auth.md [hunk 4]
```diff
@@ -358,13 +473,18 @@ providers: [
...
 
 ```
 $invoke_url/bi-authenticate?request=<request>
 ```
 
-Create a `bi-authenticate.tsx` page under `/next-auth-example/pages`. As long as your `Invoke URL` is configured properly in your [Authenticator Config](/docs/next/authentication), this is the page that will be redirected to during an authorization flow. Copy the following code snippet into that page.
+**Create a bi-authenticate route**:
+
+If you are using Next.js version 12, create a `bi-authenticate.tsx` page under `/pages/`.  
+If you are using Next.js version 13, create a `page.tsx` page under `/app/bi-authenticate/`.
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:485]:
```suggestion

```

File: docs/integration-guides/next-auth.md [hunk 5]
```diff
@@ -358,13 +473,18 @@ providers: [
...
+
+As long as your `Invoke URL` is configured properly in your [Authenticator Config](/docs/next/authentication), this is the page that will be redirected to during an authorization flow. Copy the following code snippet into that page.
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:487]:
```suggestion
  As long as your `Invoke URL` is configured properly in your [Authenticator Config](/docs/next/authentication), this is the page that will be redirected to during an authorization flow. Copy the following code snippet into that page.
```

File: docs/integration-guides/next-auth.md [hunk 6]
```diff
@@ -358,13 +473,18 @@ providers: [
 
 ### Authenticate
 
-The authenticate URL that is redirected to your application will append a `/bi-authenticate` path to your Invoke URL. Use a `/bi-authenticate` route to intercept this URL in your application:
+The authenticate URL that is redirected to your application will append a `/bi-authenticate` path to your Invoke URL. Create a `/bi-authenticate` route to intercept this URL in your application:
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:476]:
```suggestion
The authenticate URL that is redirected to your application will append a `/bi-authenticate` path to your Invoke URL. 
```

File: docs/integration-guides/next-auth.md [hunk 7]
```diff
@@ -358,13 +473,18 @@ providers: [
...
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:477]:
```suggestion

1. Create a `/bi-authenticate` route to intercept this URL in your application.
```

File: docs/integration-guides/next-auth.md [hunk 8]
```diff
@@ -358,13 +473,18 @@ providers: [
...
 
 ```
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:478]:
```suggestion
  ```
```

File: docs/integration-guides/next-auth.md [hunk 9]
```diff
@@ -358,13 +473,18 @@ providers: [
...
 $invoke_url/bi-authenticate?request=<request>
 ```
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:480]:
```suggestion
  ```
```

File: docs/integration-guides/next-auth.md [hunk 10]
```diff
@@ -358,13 +473,18 @@ providers: [
...
 
-Create a `bi-authenticate.tsx` page under `/next-auth-example/pages`. As long as your `Invoke URL` is configured properly in your [Authenticator Config](/docs/next/authentication), this is the page that will be redirected to during an authorization flow. Copy the following code snippet into that page.
+**Create a bi-authenticate route**:
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:482]:
```suggestion
2. Create a bi-authenticate route.
```

File: docs/integration-guides/next-auth.md [hunk 11]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
 </Tabs>
 
+### Configure UI
+
+NextAuth has a `useSession` hook to access session data and authentication status on the client side. In order to use this hook your components must be wrapped in a `SessionProvider` which uses React Context.
+
+1. Wrap your main components in a `SessionProvider`
+
+```javascript
+import { SessionProvider } from 'next-auth/react';
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:606]:
```suggestion
  import { SessionProvider } from 'next-auth/react';
```

File: docs/integration-guides/next-auth.md [hunk 12]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+import type { AppProps } from 'next/app';
+
+export default function App({ Component, pageProps }: AppProps) {
+  return (
+    <SessionProvider session={pageProps.session} refetchInterval={0}>
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:611]:
```suggestion
      <SessionProvider session={pageProps.session} refetchInterval={0}>
```

File: docs/integration-guides/next-auth.md [hunk 13]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      <Component {...pageProps} />
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:612]:
```suggestion
        <Component {...pageProps} />
```

File: docs/integration-guides/next-auth.md [hunk 14]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+    </SessionProvider>
+  );
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:614]:
```suggestion
    );
```

File: docs/integration-guides/next-auth.md [hunk 15]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+}
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:615]:
```suggestion
  }
```

File: docs/integration-guides/next-auth.md [hunk 16]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+```
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:616]:
```suggestion
  ```
```

File: docs/integration-guides/next-auth.md [hunk 17]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+
+2. Use the `useSession()` hook as well as `signIn` and `signOut` from `next-auth/react` inside a component.
+
+Note that `'beyondidentity'` in `signIn` is the id of the OAuth provider as configured in the providers array above.
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:620]:
```suggestion
  :::note
  `'beyondidentity'` in `signIn` is the ID of the OAuth provider as configured in the providers array above.
  :::
```

File: docs/integration-guides/next-auth.md [hunk 18]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+
+```javascript
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:622]:
```suggestion
  ```javascript
```

File: docs/integration-guides/next-auth.md [hunk 19]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+import { signIn, signOut, useSession } from 'next-auth/react';
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:623]:
```suggestion
  import { signIn, signOut, useSession } from 'next-auth/react';
```

File: docs/integration-guides/next-auth.md [hunk 20]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+const { data: session } = useSession();
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:624]:
```suggestion
  const { data: session } = useSession();
```

File: docs/integration-guides/next-auth.md [hunk 21]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+const [email, setEmail] = useState('');
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:625]:
```suggestion
  const [email, setEmail] = useState('');
```

File: docs/integration-guides/next-auth.md [hunk 22]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+
+export default function Login() {
+  const { data: session, status } = useSession();
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:628]:
```suggestion
    const { data: session, status } = useSession();
```

File: docs/integration-guides/next-auth.md [hunk 23]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+  const [email, setEmail] = useState('');
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:629]:
```suggestion
    const [email, setEmail] = useState('');
```

File: docs/integration-guides/next-auth.md [hunk 24]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+
+  const handleRegistration = () => {
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:631]:
```suggestion
    const handleRegistration = () => {
```

File: docs/integration-guides/next-auth.md [hunk 25]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+    // Call your registration API that you created above
+    // to generate an identity and bind a passkey
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:633]:
```suggestion
      // to generate an identity and bind a passkey
```

File: docs/integration-guides/next-auth.md [hunk 26]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+    callAPICreatedAbove(email);
+  };
+
+  return (
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:637]:
```suggestion
    return (
```

File: docs/integration-guides/next-auth.md [hunk 27]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+    <div>
+      {session?.user && (
+        <div>
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:640]:
```suggestion
          <div>
```

File: docs/integration-guides/next-auth.md [hunk 28]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+          <p>{`Welcome ${session.user.name}!`}</p>
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:641]:
```suggestion
            <p>{`Welcome ${session.user.name}!`}</p>
```

File: docs/integration-guides/next-auth.md [hunk 29]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+          <button onClick={() => signOut()}>Sign Out</button>
+        </div>
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:643]:
```suggestion
          </div>
```

File: docs/integration-guides/next-auth.md [hunk 30]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      )}
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:644]:
```suggestion
        )}
```

File: docs/integration-guides/next-auth.md [hunk 31]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      {!session && (
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:645]:
```suggestion
        {!session && (
```

File: docs/integration-guides/next-auth.md [hunk 32]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+        <div>
+          <div>
+            <button onClick={() => signIn('beyondidentity')}>Sign In</button>
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:648]:
```suggestion
              <button onClick={() => signIn('beyondidentity')}>Sign In</button>
```

File: docs/integration-guides/next-auth.md [hunk 33]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+          </div>
+          <div>
+            <input
+              type="email"
+              onChange={(event) => setEmail(event.target.value)}
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:653]:
```suggestion
                onChange={(event) => setEmail(event.target.value)}
```

File: docs/integration-guides/next-auth.md [hunk 34]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+            />
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:654]:
```suggestion
              />
```

File: docs/integration-guides/next-auth.md [hunk 35]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+            <button onClick={handleRegistration}>Register</button>
+          </div>
+        </div>
+      )}
+    </div>
+  );
+}
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:661]:
```suggestion
  }
```

File: docs/integration-guides/next-auth.md [hunk 36]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+```
+
+Next.js version 13 introduces React Server Components. The above assumes you are working only with client side components. To work with a server componet we can use `getServerSession`.
+
+1. Update **app/api/auth/[...nextauth]/route.ts** to export the handler so we can use this in `getServerSession`.
+
+```javascript
+const handler = NextAuth({
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:669]:
```suggestion
  const handler = NextAuth({
```

File: docs/integration-guides/next-auth.md [hunk 37]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+  providers: [
+    {
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:671]:
```suggestion
      {
```

File: docs/integration-guides/next-auth.md [hunk 38]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      id: 'beyondidentity',
+      name: 'Beyond Identity',
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:673]:
```suggestion
        name: 'Beyond Identity',
```

File: docs/integration-guides/next-auth.md [hunk 39]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      type: 'oauth',
+      wellKnown: process.env.BEYOND_IDENTITY_DISCOVERY,
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:675]:
```suggestion
        wellKnown: process.env.BEYOND_IDENTITY_DISCOVERY,
```

File: docs/integration-guides/next-auth.md [hunk 40]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      authorization: { params: { scope: 'openid' } },
+      clientId: process.env.BEYOND_IDENTITY_CLIENT_ID,
+      clientSecret: process.env.BEYOND_IDENTITY_CLIENT_SECRET,
+      idToken: true,
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:679]:
```suggestion
        idToken: true,
```

File: docs/integration-guides/next-auth.md [hunk 41]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      checks: ['state', 'pkce'],
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:680]:
```suggestion
        checks: ['state', 'pkce'],
```

File: docs/integration-guides/next-auth.md [hunk 42]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+      profile(profile) {
+        return {
+          id: profile.sub,
+          name: profile.sub,
+          email: profile.sub,
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:685]:
```suggestion
            email: profile.sub,
```

File: docs/integration-guides/next-auth.md [hunk 43]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+        };
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:686]:
```suggestion
          };
```

File: docs/integration-guides/next-auth.md [hunk 44]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+export { handler as GET, handler as POST };
+```
+
+2. In your `/app/page.tsx` use `getServerSession` to access the `session`.
+
+```javascript
+import { getServerSession } from 'next-auth/next';
+import { authOptions } from '@/app/api/auth/[...nextauth]/route';
+
+export default async function Home() {
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:701]:
```suggestion
  export default async function Home() {
```

File: docs/integration-guides/next-auth.md [hunk 45]
```diff
@@ -476,3 +596,124 @@ export default BIAuthenticate;
...
+  const session = await getServerSession(authOptions);
```
Comment (patriciamcphee) [inline docs/integration-guides/next-auth.md:702]:
```suggestion
    const session = await getServerSession(authOptions);
```

**Review by patriciamcphee — CHANGES_REQUESTED** [review]
Make sure you're following the guidelines outlined in the [style guide](https://github.com/gobeyondidentity/developer-docs/blob/docs-staging/contributor-guide/style-guide.md) and the [markdown reference guide](https://github.com/gobeyondidentity/developer-docs/blob/docs-staging/contributor-guide/markdown-reference.md).
-

**Review by patriciamcphee — CHANGES_REQUESTED** [review]
Make sure you're following the guidelines outlined in the [style guide](https://github.com/gobeyondidentity/developer-docs/blob/docs-staging/contributor-guide/style-guide.md) and the [markdown reference guide](https://github.com/gobeyondidentity/developer-docs/blob/docs-staging/contributor-guide/markdown-reference.md).
-

**Review by patriciamcphee — APPROVED** [review]

**Pull request merged**